infrastructure/ansible/roles/nginx/tasks/main.yml

```diff
@@ -4,6 +4,18 @@
     name: nginx
     state: present
 
+- name: generate dhparam.pem
+  become: true
+  shell:
+    cmd: openssl dhparam -out /etc/nginx/dhparam.pem 4096
+    creates: /etc/nginx/dhparam.pem
```

Comment:
Good usage of `creates`, it keeps the script idempotent and the 'change' and 'new' correct. It's ideal when an ansible script is run for it to actually only report changed values if something has changed.

I noticed that `/etc/nginx/dhparam.pem` is repeated a number of times in this task file. Sometimes that will be enough to justify extracting to a variable so we can be sure that everything is kept in sync. 
EG:
1) add to defaults.yml:
```
nginx_dh_param_file: /etc/nginx/dhparam.pem
```

Replace usages in tasks/main.yml to look like this:
```
dest: "{{ nginx_dh_param_file }}"
```

Note the double quotes above, those are needed otherwise the YAML parser see `dest: {` and thinks  that the value being defined is  map.

But to bring it home, because the value is also used in the nginx config, the variable becomes arguably more important:
1. ensures the file stays in sync
2. helps document that the file generated here is the same one used in configuration. We can trace the value of the variables to find usages and the correlation can then be discovered that it's a generated file, so we know where it comes from.

Reply:
In this case, the nginx config template should also contain it.

Reply:
@RoiEXLab exactly, hence the suggestion for a variable so that the value is unified. Otherwise it can be changed in one or the other and become out of sync. The usage of variables to keep config in sync is part of the reason to use something like ansible, it makes deployments very difficult to break when a value is changed (as all dependent values then are automatically updated). Hardcoded values imply the value is used in just that one place.